framework/test/unit/modules/chain/unit/api/ws/middleware/emit.js

```diff
@@ -24,45 +23,44 @@ describe('emitMiddleware', () => {
 	let validReq;
 	let validNext;
 
-	beforeEach(done => {
+	beforeEach(async () => {
 		emitMiddleware(validReq, validNext);
-		done();
+	});
+
+	afterEach(async () => {
+		sinonSandbox.restore();
 	});
 
 	describe('when valid req and next params provided', () => {
 		const validSocketId = 0;
-		before(done => {
+		before(async () => {
 			validReq = {
 				data: {},
 				socket: {
 					id: validSocketId,
 				},
 			};
-			validNext = sinon.spy();
-			done();
+			validNext = sinonSandbox.spy();
+			connectionsTable.getNonce = sinonSandbox.stub();
 		});
 
-		afterEach(() => validNext.resetHistory());
+		afterEach(async () => {
+			validNext.resetHistory();
+			validReq.data = {};
+			connectionsTable.getNonce.reset();
+		});
 
 		it('should call validNext', async () => expect(validNext).calledOnce);
 
-		describe('when req.event is one of the events responsible for receiving data on P2P layer', () => {
-			const receiveDataEvents = [
-				'postBlock',
-				'postTransactions',
-				'postSignatures',
-			];
-			receiveDataEvents.forEach(() => {
-				before(done => {
-					connectionsTable.getNonce = sinon.stub();
-					validReq.event = 'postBlock';
-					done();
-				});
-
-				afterEach(done => {
-					connectionsTable.getNonce.reset();
-					validReq.data = {};
-					done();
+		const receiveDataEvents = [
```

Comment:
Can this declaration be moved to near the top of the `describe` block, it looks a bit confusing here.